sdk/eventhub/event-hubs/src/batchingReceiver.ts

```diff
@@ -110,6 +119,7 @@ export class BatchingReceiver extends EventHubReceiver {
           enqueuedTimeUtc: data.enqueuedTimeUtc,
           partitionKey: data.partitionKey
         };
+        this._checkpoint = receivedEventData.sequenceNumber!;
```

Comment:
Regarding the ! here...

Just curious... do we know of a case where sequenceNumber is undefined/null on `ReceivedEventData`?
Previously when we used the same `EventData` class for sending and receiving, it was possible to have sequenceNumber, offset and enqueuedTime as undefined/null as these won't be set on `EventData` when sending.
Now that we have a separate interface for `ReceivedEventData`, should these properties remain "optional"?

Reply:
We separated the send `EventData` and `ReceivedEventData`, so properties inside `ReceivedEventData` shouldn't be optional.

I'll update the `ReceivedEventData` properties in same PR, next time we will create separate issues for these kind of code clean ups  :)

Reply:
> I'll update the ReceivedEventData properties in same PR, next time we should create separate issues for these kind of code clean ups :)

agreed. I am ok with having a separate PR even right now.